### ISSUE TITLE
On Firefox @Prop doesn't update a runtime 

### ISSUE THREAD

**Issue by giotramu** [ionic-team/stencil#511]
**Stencil version:**

```
 @stencil/core@0.4.0
```

**I'm submitting a:**

- [X] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or https://stencil-worldwide.slack.com

**Current behavior:**
On Firefox 58.0.2 and on Firefox Developer Edition 59.0b7 I can't update @Prop at run-time.
It doesn’t seem to enter in the @Watch decorator callback, indeed it doesn't print the console.log message 'Name changed'.
I show bug details in screen-capture: https://ufile.io/g2bir

**Expected behavior:**
When I update attribute on custom-element I expect that changes are reflectend in local-DOM.

**Steps to reproduce:**
1 - npm install
2 - npm dev --es5

**Related code:**

```tsx
import {Component, Method, Prop, State, Watch} from '@stencil/core';

@Component({
  shadow: true,
  styleUrl: 'uxd-collapsible.pcss',
  tag: 'uxd-collapsible'
})
export class UXDCollapsible {
  @Prop() public name: string;

  @State() private collapsed: boolean = true;
  @State() private handlerTitle: string;

  @Method()
  public toggle(): void {
    this.collapsed = !this.collapsed;
  }

  @Watch('name')
  private onNameChange(newVal: string): void {
    console.log('Name changed');
    this.handlerTitle = newVal;
  }

  private componentWillLoad(): void {
    this.handlerTitle = this.name;
  }

  private render(): JSX.Element {
    return (
      <div>
        <header onClick={this.toggle.bind(this)}>
          <span>{this.handlerTitle}</span>
        </header>
        <main hidden={this.collapsed}>
          <slot />
        </main>
      </div>
    );
  }
}
```

```html
<uxd-collapsible name="UXD Elements - Collapsible Panel">
  <ul>
    <li>Test content 1</li>
    <li>Test content 2</li>
  </ul>
</uxd-collapsible>
```

**Other information:**
I posted the problem also in stencil-dev Slack channel:
https://stencil-worldwide.slack.com/archives/C944RNLRX/p1518082931000340
The problem seems related to my machine. My setup: OSX 10.13.3 and Node v9.5.0

**Comment by jgw96** [comment]
Hmm @giotramu could you post your repo that i could clone and try? I just tried your example component locally and its working fine for me on the latest firefox stable.

**Comment by giotramu** [comment]
@jgw96 very strange. Can you check here: https://gitlab.com/giotramu/uxd-elements ?
Have you tested in Firefox Developer Edition, Classic or Focus?

**Comment by jgw96** [comment]
Hey @giotramu , so to isolate the issue i made a small component that uses `Watch` to know when a prop updates. In firefox nightly and the current stable version of Firefox it works fine. Only difference is im on `@stencil/core0.5.0-0`. Mind updating to that version and giving it a try?

**Comment by jgw96** [comment]
Hello all! As it seems it has been a while since there was any activity on this issue I will be closing it for now. Feel free to comment if you are still running into this issue. Thanks for using Stencil!

**Comment by julesjanssen** [comment]
Just FYI: I think it depends on how you change the attribute. If you do `element.name = 'xxx'` it will work, but if you use `element.setAttribute('name', 'xxx')` it doesn't trigger the change event.

**Comment by silentHoo** [comment]
@jgw96 Could you add a note into the docs that calling `element.setAttribute('name', 'xxx')` will not trigger the change event in Firefox as @julesjanssen mentioned above?

And 2) to note: if you have a camelCase property, you cannot use `element['camel-case'] = 'xxx'`, you have to use `element.camelCase = 'xxx'`.

**Comment by julesjanssen** [comment]
I think this was fixed in the meantime, see #623. It works for me with the 0.7.1 release.

Not sure if the syntax you use in point 2 is valid at all.
